ANTLR/src/Compiler/Lang.g4

```diff
@@ -19,16 +19,10 @@ stmt : expr SEMICOLON ;
 expr : val                                                                    #value
      | op=NEGATE val                                                          #uneryOp
      | left=expr op=POW right=expr                                            #binaryOp
-     | left=expr op=DIVIDE right=expr                                         #binaryOp
-     | left=expr op=MULTIPLY right=expr                                       #binaryOp
-     | left=expr op=PLUS right=expr                                           #binaryOp
-     | left=expr op=MINUS right=expr                                          #binaryOp
-     | left=expr op=LOGEQ right=expr                                          #binaryOp
-     | left=expr op=LOGNOTEQ right=expr                                       #binaryOp
-     | left=expr op=LOGLESS right=expr                                        #binaryOp
-     | left=expr op=LOGGREATER right=expr                                     #binaryOp
-     | left=expr op=LOGLESSOREQ right=expr                                    #binaryOp
-     | left=expr op=LOGGREATEROREQ right=expr                                 #binaryOp
+     | left=expr op=(DIVIDE|MULTIPLY) right=expr                              #binaryOp
+     | left=expr op=(PLUS|MINUS) right=expr                                   #binaryOp
+     | left=expr op=LOGRELATION right=expr                                    #binaryOp
+     | left=expr op=LOGEQUALITY right=expr                                    #binaryOp
```

Comment:
This one is better than the `LOGRELATION` as the name of the "variable" is more fitting. But for the sake of consistency, I'd prefer it like this: 
```suggestion
     | left=expr op=(LOGEQ | LOGNOTEQ) right=expr                                    #binaryOp
```